CMakeLists.txt

```diff
@@ -99,7 +99,8 @@ option(ENABLE_TEST "Enables unit testing" ON)
 option(ENABLE_PROFILING "Enables profiling support in the application" OFF)
 option(ENABLE_COVERAGE "Enable building and installation of examples" OFF)
 option(ENABLE_EXAMPLES "Enable building and installation of examples" OFF)
-option(ENABLE_DOC "Enable building and installation of documentation" OFF)
+option(ENABLE_USER_DOC "Enable building and installation of the user documentation" OFF)
+option(ENABLE_API_DOC "Enable building and installation of the api documentation" OFF)
```

Comment:
This would require that we bump the CMake version. Currently we use 2.6 which is really old and does not seem to support the `cmake_dependant_option` macro.

I agree that it would add readability by keeping the doc guard. I feel that we should maybe try to split the CMake files for the user doc and api doc as well. The problem with that is the doxygen will require a separate folder which might be a bit much for 2 files.